src/mpid/ch4/netmod/ofi/ofi_win.h

```diff
@@ -16,32 +16,30 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_win_do_progress(MPIR_Win * win)
     int mpi_errno = MPI_SUCCESS;
     int itercount = 0;
     int ret;
-    uint64_t tcount, donecount;
+    uint64_t donecount;
     MPIDI_OFI_win_request_t *r;
 
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_OFI_WIN_DO_PROGRESS);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_OFI_WIN_DO_PROGRESS);
 
     while (1) {
-        tcount = *MPIDI_OFI_WIN(win).issued_cntr;
         donecount = fi_cntr_read(MPIDI_OFI_WIN(win).cmpl_cntr);
 
-        MPIR_Assert(donecount <= tcount);
+        MPIR_Assert(donecount <= *MPIDI_OFI_WIN(win).issued_cntr);
 
-        while (tcount > donecount) {
-            MPIR_Assert(donecount <= tcount);
+        while (*MPIDI_OFI_WIN(win).issued_cntr > donecount) {
+            MPIR_Assert(donecount <= *MPIDI_OFI_WIN(win).issued_cntr);
             MPIDI_OFI_PROGRESS(0);
```

Comment:
The change seems unnecessarily large. Can we just add one line to update `tcount` after `MPIDI_OFI_PROGRESS(0)` ? Please add a comment that `tcount` may be updated in `MPIDI_OFI_PROGRESS(0)` if there is any incoming AM and triggers RDMA get. Thus, we need load it again.

Reply:
Can you add brief description of this issue and the fix in commit message too?